### ISSUE TITLE
<mj-style> invalid in LV 1.0 heex compiler

### ISSUE THREAD

**Issue by ftes** [phoenixframework/phoenix_live_view#3550]
### Environment

* Elixir version (elixir -v):
  * Erlang/OTP 27 [erts-15.1] [source] [64-bit] [smp:14:14] [ds:14:14:10] [async-threads:1] [jit]
  * Elixir 1.17.3 (compiled with Erlang/OTP 27)
* Phoenix version (mix deps): 1.7
* Phoenix LiveView version (mix deps): main branch
* Operating system: macos
* Browsers you attempted to reproduce this bug on (the more the merrier): brave
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: yes

### Actual behavior

<details>
<summary>Single file app</summary>

```ex
Application.put_env(:sample, Example.Endpoint,
  http: [ip: {127, 0, 0, 1}, port: 5001],
  server: true,
  live_view: [signing_salt: "aaaaaaaa"],
  secret_key_base: String.duplicate("a", 64)
)

Mix.install([
  {:plug_cowboy, "~> 2.5"},
  {:jason, "~> 1.0"},
  {:phoenix, "~> 1.7"},
  # please test your issue using the latest version of LV from GitHub!
  {:phoenix_live_view, github: "phoenixframework/phoenix_live_view", branch: "main", override: true}
])

# build the LiveView JavaScript assets (this needs mix and npm available in your path!)
path = Phoenix.LiveView.__info__(:compile)[:source] |> Path.dirname() |> Path.join("../")
System.cmd("mix", ["deps.get"], cd: path, into: IO.binstream())
System.cmd("npm", ["install"], cd: Path.join(path, "./assets"), into: IO.binstream())
System.cmd("mix", ["assets.build"], cd: path, into: IO.binstream())

defmodule Example.ErrorView do
  def render(template, _), do: Phoenix.Controller.status_message_from_template(template)
end

defmodule Example.HomeLive do
  @moduledoc false
  use Phoenix.LiveView, layout: {__MODULE__, :live}

  def mount(_params, _session, socket) do
    {:ok, assign(socket, :count, 0)}
  end

  def render("live.html", assigns) do
    ~H"""
    <script src="/assets/phoenix/phoenix.js">
    </script>
    <script src="/assets/phoenix_live_view/phoenix_live_view.js">
    </script>
    <%!-- uncomment to use enable tailwind --%>
    <%!-- <script src="https://cdn.tailwindcss.com"></script> --%>
    <script>
      let liveSocket = new window.LiveView.LiveSocket("/live", window.Phoenix.Socket)
      liveSocket.connect()
    </script>
    <style>
      * { font-size: 1.1em; }
    </style>
    <mj-style>
      * { font-size: 1.1em; }
    </mj-style> {@inner_content}
    """
  end

  def render(assigns) do
    ~H"""
    {@count} <button phx-click="inc">+</button>
    <button phx-click="dec">-</button>
    """
  end

  def handle_event("inc", _params, socket) do
    {:noreply, assign(socket, :count, socket.assigns.count + 1)}
  end

  def handle_event("dec", _params, socket) do
    {:noreply, assign(socket, :count, socket.assigns.count - 1)}
  end
end

defmodule Example.Router do
  use Phoenix.Router

  import Phoenix.LiveView.Router

  pipeline :browser do
    plug(:accepts, ["html"])
  end

  scope "/", Example do
    pipe_through(:browser)

    live("/", HomeLive, :index)
  end
end

defmodule Example.Endpoint do
  use Phoenix.Endpoint, otp_app: :sample

  socket("/live", Phoenix.LiveView.Socket)

  plug Plug.Static, from: {:phoenix, "priv/static"}, at: "/assets/phoenix"
  plug Plug.Static, from: {:phoenix_live_view, "priv/static"}, at: "/assets/phoenix_live_view"

  plug(Example.Router)
end

{:ok, _} = Supervisor.start_link([Example.Endpoint], strategy: :one_for_one)
Process.sleep(:infinity)
```

</details>

Input:
```heex
<!-- layout.mjml.heex -->
<mj-style>
  * { font-size: 1.1em; }
</mj-style>
```

Output:
```ex
** (SyntaxError) invalid syntax found on main.exs:51:21:
    error: invalid character "e" after number 1.1. If you intended to write a number, make sure to separate the number from the character (using comma, space, etc). If you meant to write a function name or a variable, note that identifiers in Elixir cannot start with numbers. Unexpected token: e
    │
 51 │          font-size: 1.1em;
    │                     ^
    │
    └─ main.exs:51:21
    (phoenix_live_view 1.0.0) lib/phoenix_live_view/tag_engine.ex:448: Phoenix.LiveView.TagEngine.handle_token/2
    (elixir 1.17.3) lib/enum.ex:2531: Enum."-reduce/3-lists^foldl/2-0-"/3
    (phoenix_live_view 1.0.0) lib/phoenix_live_view/tag_engine.ex:209: Phoenix.LiveView.TagEngine.handle_body/1
    (phoenix_live_view 1.0.0) expanding macro: Phoenix.Component.sigil_H/2
    main.exs:36: Example.HomeLive.render/2
```

### Expected behavior
Support style in `<mj-style>` tag, like it was supported pre 1.0.
Basically, allow the same content in `<mj-style>` as is allowed in `<style>`.

I assume this special handling is to blame:
https://github.com/phoenixframework/phoenix_live_view/blob/2e25dcc8a5454d4d4f7453a2a1eef3ee64e2eb0f/lib/phoenix_live_view/tokenizer.ex#L119

**Comment by ftes** [comment]
Workaround:

```heex
{Phoenix.HTML.raw("""
  <mj-style>
    * { font-size: 1.1em; }
  </mj-style>
"""}
```

**Comment by andrewtimberlake** [comment]
The solution is to use the `phx-no-curly-interpolation` attribute
```html
<mj-style phx-no-curly-interpolation>
  * { font-size: 1.1em; }
</mj-style>
```

**Comment by ftes** [comment]
Thanks @andrewtimberlake, I wasn't aware that was a thing.

https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#sigil_H/2-interpolating-blocks

That resolves it.
